pkg/client/client.go

```diff
@@ -352,6 +385,146 @@ func (c *serviceImpl) doRequest(method, path string, reqpl, respl any) error {
 	return nil
 }
 
+const openConnections = 10
+
+type filesUploader struct {
+	Client         *http.Client
+	EndpointURL    string
+	Filenames      []string
+	ChunkSizeBytes int64
+	DumpOut        io.Writer
+}
+
+type fileInfo struct {
+	Name        string
+	TotalChunks int
+}
+
+func (u *filesUploader) Upload() error {
+	var totalChunks int
+	var infos []fileInfo
+	for _, name := range u.Filenames {
+		stat, err := os.Stat(name)
+		if err != nil {
+			return nil
+		}
+		info := fileInfo{
+			Name:        name,
+			TotalChunks: int(math.Ceil(float64(stat.Size()) / float64(u.ChunkSizeBytes))),
```

Comment:
Use this formula with integer operations instead: `(file_size + chunk_size - 1) / chunk_size` 
Floating point operations are slow and approximate so it can introduce errors (though very unlikely in this case). The downside of the integer only operations is that the addition could potentially overflow, but that's not an issue for our use case with `int64`.

Reply:
Good point! done